### ISSUE TITLE
Stuck on "waiting for connection"

### ISSUE THREAD

**Issue by SifuDaigo** [jheling/freeathome#34]
Whenever I enable the configuration for freeathome in my config-file HA gets stuck in an everlasting `waiting for connection` and the frontend won't load up. I already tried multiple times and even installed a fresh installation of homeassistant, but I can't get it to work.

Thanks for the effort!!!

**Comment by dbrouwer1979** [comment]
Just try to reboot several times. Eventually it will load.

**Comment by SifuDaigo** [comment]
Thank you for the quick response. I already rebooted about ~15 times - sadly always the same result.

**Comment by dbrouwer1979** [comment]
Maybe try this: disable freeathome. Then reboot. Then enable freeathome in config and reboot only home assistant from frontend  (settings - server management - server management - reboot)

**Comment by SifuDaigo** [comment]
I did that just now - same probleme still.

**Comment by bartdorlandt** [comment]
This is my lazy script to work around it for now...

Dependency: 'nc'
I'm using HA with docker. Feel free to change it to your liking.

```
#!/usr/bin/env bash

HOST=server
PORT=8123
SUCCESS=1
SLP=10
HARESTART='/usr/bin/docker restart home-assistant'

while [ $SUCCESS -ne 0 ]; do
    echo "Testing connectivity."
    /usr/bin/nc -z $HOST $PORT
    SUCCESS=$?

    if [[ $SUCCESS == 0 ]] ; then
        echo "Success!!"
        echo "    HA + Free@home are reachable"
    else
        echo "Restarting home-assistant"
        $HARESTART
        echo "waiting $SLP seconds and will test again."
        sleep $SLP
    fi
done
```

**Comment by jheling** [comment]
In the issue 'TypeError: 'NoneType' object is not an iterator #27' I posted a solution that work for me.  The 'waiting for connection' is a result of the same bug. If it works for you, can you send a reply?

**Comment by apfelmaennchen** [comment]
For me a _UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 1231: ordinal not in range(128)_ error caused the "waiting for connection" loop.

As a solution I changed in freeathome/freeathome/fah/crypto.py line 138 
from: 
`result_ascii = result.decode('ascii')`
to: 
`result_ascii = result.decode('utf-8')`

I don't know whether this has side effects or not. At the moment it solved the problem for me.

**Comment by miezie** [comment]
Makes sense... how are the different encoding settings? On both FAH and a given HA instance?

**Comment by jheling** [comment]
The encoding should be utf-8. I have corrected crypto.py.  The issue 'TypeError: 'NoneType' object is not an iterator is resolved in  #27.  So i am closing this issue.
